src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp

```diff
@@ -0,0 +1,429 @@
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Douglas McCloskey, Pasquale Domenico Colaianni $
+// $Authors: Douglas McCloskey, Pasquale Domenico Colaianni $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.h>
+
+namespace OpenMS
+{
+  TargetedSpectraExtractor::TargetedSpectraExtractor() :
+    DefaultParamHandler("TargetedSpectraExtractor")
+  {
+    getDefaultParameters(defaults_);
+    defaultsToParam_(); // write defaults into Param object param_
+  }
+
+  TargetedSpectraExtractor::~TargetedSpectraExtractor() {}
+
+  void TargetedSpectraExtractor::updateMembers_()
+  {
+    rt_unit_ = (String)param_.getValue("rt_unit");
+    rt_window_ = (double)param_.getValue("rt_window");
+    min_score_ = (double)param_.getValue("min_score");
+    min_forward_match_ = (double)param_.getValue("min_forward_match");
+    min_reverse_match_ = (double)param_.getValue("min_reverse_match");
+    mz_tolerance_ = (double)param_.getValue("mz_tolerance");
+    mz_unit_is_Da_ = param_.getValue("mz_unit_is_Da").toBool();
+
+    sgolay_frame_length_ = (UInt)param_.getValue("sgolay_frame_length");
+    sgolay_polynomial_order_ = (UInt)param_.getValue("sgolay_polynomial_order");
+    gauss_width_ = (double)param_.getValue("gauss_width");
+    use_gauss_ = param_.getValue("use_gauss").toBool();
+    signal_to_noise_ = (double)param_.getValue("signal_to_noise");
+
+    peak_height_min_ = (double)param_.getValue("peak_height_min");
+    peak_height_max_ = (double)param_.getValue("peak_height_max");
+    fwhm_threshold_ = (double)param_.getValue("fwhm_threshold");
+
+    tic_weight_ = (double)param_.getValue("tic_weight");
+    fwhm_weight_ = (double)param_.getValue("fwhm_weight");
+    snr_weight_ = (double)param_.getValue("snr_weight");
+  }
+
+  void TargetedSpectraExtractor::getDefaultParameters(Param& params)
```

Comment:
please check this method again.
e.g.: rt_window still has written "seconds" in the description.

I personally still think that it would make sense to "forward" the parameters from the underlying algorithms for better maintainability. But I leave that to you to decide.

Reply:
Also, consider looking at other tools parameter names. It would be nicer to make these consistent throughout OpenMS.

Description is - in parts - a bit short and could be extended (e.g., what does minimum forward match mean? what is the unit? Is it just a score? etc.)